### PR TITLE
downloader-aware scheduler cleanups

### PR DESCRIPTION
**Pull request by kmike**
Changes are rather mechanical so far, I haven't got to more complex parts yet.

### CODE REVIEW

File: scrapy/pqueues.py [hunk 1]
```diff
@@ -53,43 +46,28 @@ def _scheduler_slot(request):
...
-
-    def __eq__(self, other):
-        return (self.priority, self.slot) == (other.priority, other.slot)
-
-    def __lt__(self, other):
-        return (self.priority, self.slot) < (other.priority, other.slot)
-
+class PrioritySlot(namedtuple("PrioritySlot", ["priority", "slot"])):
+    """ ``(priority, slot)`` tuple which uses a path-safe slot name
+    when converting to str """
```
Comment (whalebot-helmsman) [inline scrapy/pqueues.py:67]: Based on https://stackoverflow.com/a/28568351/8640733 we need `__slots__ = ()` here

File: scrapy/pqueues.py [hunk 2]
```diff
@@ -53,43 +46,28 @@ def _scheduler_slot(request):
         url = request.get('url', None)
         slot = urlparse(url).hostname or ''
     elif isinstance(request, Request):
-        url = request.url
         slot = urlparse_cached(request).hostname or ''
 
+    # FIXME: meta is not modified when request is a dict and no meta is stored?
```
Comment (whalebot-helmsman) [inline scrapy/pqueues.py]: To address this concern we should preserve https://github.com/whalebot-helmsman/scrapy/blob/d8e2b25ba7811543d4be6c6bb6a8e851653bf368/scrapy/pqueues.py#L40 and replace `meta = request.get('meta', dict())` by `meta = request.setdefault('meta', dict())` inside this code

File: scrapy/pqueues.py [hunk 3]
```diff
@@ -167,10 +132,10 @@ class DownloaderAwarePriorityQueue(SlotBasedPriorityQueue):
     _DOWNLOADER_AWARE_PQ_ID = 'DOWNLOADER_AWARE_PQ_ID'
 
     @classmethod
-    def from_crawler(cls, crawler, qfactory, startprios={}):
+    def from_crawler(cls, crawler, qfactory, startprios=None):
```
Comment (whalebot-helmsman) [inline scrapy/pqueues.py:136]: Sorry, I don't understand why you replace `{}` by `None` here. `{}` was used in accord with https://github.com/scrapy/queuelib/blob/574ef504d217993dd5c098068ad43852868b5293/queuelib/pqueue.py#L24

**Review by whalebot-helmsman — COMMENTED** [review]
I leave comments on small improvements possible here and there

File: scrapy/pqueues.py (hunk 3 above)
Comment (kmike) [reply scrapy/pqueues.py:136]: it is just a style change - mutable default arguments are considered bad style and a source of possible errors; in pqueues it was different, as tuple is immutable. None works the same here, if I'm not mistaken.

**Comment by whalebot-helmsman** [comment]
Thanks @kmike

**Pull request merged**